tests/_support/AcceptanceTester.php

```diff
@@ -12,15 +15,226 @@
+	}
+
+	/**
+	 * Function to check for PHP Notices or Warnings
+	 *
+	 * @param string $page Optional, if not given checks will be done in the current page
+	 *
+	 * @note: doAdminLogin() before
+	 */
+	public function checkForPhpNoticesOrWarnings($page = null)
+	{
+		$I = $this;
+
+		if ($page)
+		{
+			$I->amOnPage($page);
+		}
+
+		$I->dontSeeInPageSource('Notice:');
+		$I->dontSeeInPageSource('<b>Notice</b>:');
+		$I->dontSeeInPageSource('Warning:');
+		$I->dontSeeInPageSource('<b>Warning</b>:');
+		$I->dontSeeInPageSource('Strict standards:');
+		$I->dontSeeInPageSource('<b>Strict standards</b>:');
+		$I->dontSeeInPageSource('The requested page can\'t be found');
+	}
+
+	/**
+	 * Function to select Toolbar buttons in Joomla! Admin Toolbar Panel
+	 *
+	 * @param   string  $button  The full name of the button
+	 *
+	 * @return  void
+	 */
+	public function clickToolbarButton($button)
+	{
+		$I = $this;
+		$input = strtolower($button);
+
+		// @todo Needs to find way to work with different window size.
+		$screenSize = explode("x", $this->getConfiguration('window_size'));
+
+		if ($screenSize[0] <= 480)
+		{
+			$I->click('Toolbar');
+		}
+
+		switch ($input)
+		{
+			case "new":
+				$I->click(['xpath' => "//div[@id='toolbar-new']//button"]);
+				break;
+			case "edit":
+				$I->click(['xpath' => "//div[@id='toolbar-edit']//button"]);
+				break;
+			case "publish":
+				$I->click(['xpath' => "//div[@id='toolbar-publish']//button"]);
+				break;
+			case "unpublish":
+				$I->click(['xpath' => "//div[@id='toolbar-unpublish']//button"]);
+				break;
+			case "archive":
+				$I->click(['xpath' => "//div[@id='toolbar-archive']//button"]);
+				break;
+			case "check-in":
+				$I->click(['xpath' => "//div[@id='toolbar-checkin']//button"]);
+				break;
+			case "batch":
+				$I->click(['xpath' => "//div[@id='toolbar-batch']//button"]);
+				break;
+			case "rebuild":
+				$I->click(['xpath' => "//div[@id='toolbar-refresh']//button"]);
+				break;
+			case "trash":
+				$I->click(['xpath' => "//div[@id='toolbar-trash']//button"]);
+				break;
+			case "save":
+				$I->click(['xpath' => "//div[@id='toolbar-apply']//button"]);
+				break;
+			case "save & close":
+				$I->click(['xpath' => "//div[@id='toolbar-save']//button"]);
+				break;
+			case "save & new":
+				$I->click(['xpath' => "//div[@id='toolbar-save-new']//button"]);
+				break;
+			case "cancel":
+				$I->click(['xpath' => "//div[@id='toolbar-cancel']//button"]);
+				break;
+			case "options":
+				$I->click(['xpath' => "//div[@id='toolbar-options']//button"]);
+				break;
+			case "empty trash":
+				$I->click(['xpath' => "//div[@id='toolbar-delete']//button"]);
+				break;
+			case "unblock":
+				$I->click(['xpath' => "//div[@id='toolbar-unblock']//button"]);
+				break;
+			case "delete":
+				$I->click(['xpath' => "//div[@id='toolbar-delete']//button"]);
+				break;
+			case "featured":
+				$I->click(['xpath' => "//div[@id='toolbar-featured']//button"]);
+				break;
+		}
+	}
```

Comment:
Functions like the previous one will be moved to step or helper files. But since we will be improving the current code through iterations will merge.